### PR TITLE
Building Blocks selector

### PR DESCRIPTION
**Pull request by alfonsograziano**
### Ticket:

- [16814](https://dev.azure.com/OGCIO-Digital-Services/Digital%20Services%20Programme/_workitems/edit/16814)

### Description

Creating a new building blocks slector

## Type

- [ ] **Dependency upgrade**
- [ ] **Bug fix**
- [X] **New feature**
- [ ] **Dev change**

### Screenshots:
Screenshot 2024-05-15 at 12 37 28

### CODE REVIEW

**Comment by alfonsograziano** [comment]
For now, the selector has been implemented in the Payments building block application, but ideally, we would have a library of shared components across all the applications to make it reusable. Perhaps temporarily we could use the design-system package and expose everything from there? But this also has some challenges, happy to hear your opinions @SamSalvatico @peschina @nnorbert @Thurfjell

File: apps/payments/.env.sample [hunk 1]
```diff
@@ -27,3 +27,8 @@ LOGTO_ENDPOINT=http://localhost:3301/
 LOGTO_APP_SECRET=logto_app_secret
 LOGTO_COOKIE_SECRET=logto_cookie_secret
 LOGTO_BASE_URL=http://localhost:3001
+
+NEXT_PUBLIC_MESSAGING_SERVICE_ENTRY_POINT = http://localhost:3002
+NEXT_PUBLIC_PAYMENTS_SERVICE_ENTRY_POINT = http://localhost:3001
+NEXT_PUBLIC_LIFE_EVENTS_SERVICE_ENTRY_POINT = http://localhost:3000
+NEXT_PUBLIC_BUILDING_BLOCKS_LANDING_PAGE = https://blocks.gov.ie/
```
Comment (peschina) [inline apps/payments/.env.sample:34]: Newline please :)

File: apps/payments/app/[locale]/(hosted)/buildingBlocksTiles/TilesButton.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import { useState, useEffect } from "react";
+import ds from "design-system/";
+import { ComponentProps } from "react";
+import { useTranslations } from "next-intl";
+
+const buildingBlocksMainLandingPage =
+  (process.env.NEXT_PUBLIC_BUILDING_BLOCKS_LANDING_PAGE as string) ?? "#";
+const services: TileProps[] = [
+  {
+    url: process.env.NEXT_PUBLIC_PAYMENTS_SERVICE_ENTRY_POINT as string,
```
Comment (peschina) [inline apps/payments/app/[locale]/(hosted)/buildingBlocksTiles/TilesButton.tsx:11]: shall we have a fallback for all these vars or throw an error if they are missing?

**Review by peschina — APPROVED** [review]
LGTM, a few minor comments. We'll need to iterate back to make it responsive once the header is responsive too

**Review by williamwd — APPROVED** [review]

**Pull request merged**